src/test/java/com/researchspace/api/v1/controller/SamplesApiControllerMVCIT.java

```diff
@@ -1083,4 +1080,62 @@ public void changeSampleOwner() throws Exception {
     assertNotNull(editedSample);
     assertEquals(anotherUser.getUsername(), editedSample.getOwner().getUsername());
   }
+
+  @Test
+  public void createSampleFromTemplateWithValidTimeIsAccepted() throws Exception {
+    User anyUser = createInitAndLoginAnyUser();
+    String apiKey = createApiKeyForuser(anyUser);
+
+    Long sampleTemplateId = getComplexSampleTemplate(anyUser).get().getId();
+    String validTime = "10:15";
+    String sampleWithExtraFieldsJSON =
+        makeJsonSampleWithIdAndTime(sampleTemplateId.toString(), validTime);
+
+    MvcResult result =
+        this.mockMvc
+            .perform(
+                createBuilderForPostWithJSONBody(
+                    apiKey, "/samples", anyUser, sampleWithExtraFieldsJSON))
+            .andExpect(status().isCreated())
+            .andReturn();
+
+    ApiSampleWithFullSubSamples createdSample =
+        mvcUtils.getFromJsonResponseBody(result, ApiSampleWithFullSubSamples.class);
+
+    assertEquals(validTime, createdSample.getExtraFields().get(0).getContent());
+  }
+
+  @Test
+  public void createSampleFromTemplateWithInvalidTimeIsRejected() throws Exception {
+    User anyUser = createInitAndLoginAnyUser();
+    String apiKey = createApiKeyForuser(anyUser);
+
+    Long sampleTemplateId = getComplexSampleTemplate(anyUser).get().getId();
+    String invalidTime = "9:15"; // time format should be 24 hours with 4 digits e.g. 09:15
+    String sampleWithExtraFieldsJSON =
+        makeJsonSampleWithIdAndTime(sampleTemplateId.toString(), invalidTime);
+
+    MvcResult result =
+        this.mockMvc
+            .perform(
+                createBuilderForPostWithJSONBody(
+                    apiKey, "/samples", anyUser, sampleWithExtraFieldsJSON))
+            .andExpect(status().isBadRequest())
+            .andReturn();
+
+    String responseText = result.getResponse().getContentAsString();
+    String expectedErrorMessage = "9:15 is an invalid 24hour time format. Valid format is 00:00.";
+    assertTrue(responseText.contains(expectedErrorMessage));
+  }
+
+  private String makeJsonSampleWithIdAndTime(String id, String time) {
+    return "{ \"templateId\": \""
+        + id
+        + "\",\"fields\": [ { \"content\": \"3.14\" }, {}, {},  { \"type\": \"Text\","
+        + " \"content\": \"text content\" }, {}, {}, {}, {}, {}], \"extraFields\" : [   {"
+        + " \"name\": \"extraFieldName\", \"type\" : \"time\", \"content\": \""
+        + time
+        + "\" } ],"
+        + " \"expiryDate\":null, \"name\": \"sample1\" }";
+  }
```

Comment:
as discussed on rspace-core-model PR, this test should actually try setting an invalid _field_ of "Time" type, rather than _extraField_ of such type. For that it must use a Sample Template that has a Time value, e.g. ComplexSampleTemplate created in dev/run environment....

... only that looking at [RecordFactory](https://github.com/ResearchSpace-ELN/rspace-core-model/blob/a08e8be5c1e7cf41887f2cb97c9055d165f0ac6e/src/main/java/com/researchspace/model/record/RecordFactory.java#L328) class I see that "CompexSampleTemplate" doesn't have Time field. That must be by accident, it has all the other field types. It would be good to extend the template to have Time field too, it'll make it easier to test the functionality in unit test but also on localhost UI.

Reply:
Thanks, this makes sense. I've added the additional time field to the `complexSampleTemplate` and updated existing tests.